main.go

```diff
@@ -127,34 +130,52 @@ func main() {
 	kubeClient, err := kubernetes.NewForConfig(kubeConfig)
 	exitOnError(err, "unable to create Kubernetes client")
 
-	exitOnError(loadIntoOrCreate(ctx, kubeClient, namespaceOrDie(), configMapName, cfg), "unable to initialise configuration")
+	namespace := namespaceOrDie()
 
-	kubeConfig.Burst = int(pointer.Int32Deref(cfg.ClientConnection.Burst, int32(rest.DefaultBurst)))
-	kubeConfig.QPS = pointer.Float32Deref(cfg.ClientConnection.QPS, rest.DefaultQPS)
+	exitOnError(loadIntoOrCreate(ctx, kubeClient, namespace, configMapName, cfg), "unable to initialise configuration")
+
+	kubeConfig.Burst = int(ptr.Deref(cfg.ClientConnection.Burst, int32(rest.DefaultBurst)))
+	kubeConfig.QPS = ptr.Deref(cfg.ClientConnection.QPS, rest.DefaultQPS)
 	setupLog.V(2).Info("REST client", "qps", kubeConfig.QPS, "burst", kubeConfig.Burst)
 
 	mgr, err := ctrl.NewManager(kubeConfig, ctrl.Options{
-		Scheme:                     scheme,
-		MetricsBindAddress:         cfg.Metrics.BindAddress,
+		Scheme: scheme,
+		Metrics: metricsserver.Options{
+			BindAddress: cfg.Metrics.BindAddress,
+		},
 		HealthProbeBindAddress:     cfg.Health.BindAddress,
-		LeaderElection:             pointer.BoolDeref(cfg.LeaderElection.LeaderElect, false),
+		LeaderElection:             ptr.Deref(cfg.LeaderElection.LeaderElect, false),
 		LeaderElectionID:           cfg.LeaderElection.ResourceName,
 		LeaderElectionNamespace:    cfg.LeaderElection.ResourceNamespace,
 		LeaderElectionResourceLock: cfg.LeaderElection.ResourceLock,
 		LeaseDuration:              &cfg.LeaderElection.LeaseDuration.Duration,
 		RetryPeriod:                &cfg.LeaderElection.RetryPeriod.Duration,
 		RenewDeadline:              &cfg.LeaderElection.RenewDeadline.Duration,
 	})
-	exitOnError(err, "unable to start manager")
+	exitOnError(err, "unable to create manager")
+
+	certsReady := make(chan struct{})
+	exitOnError(setupCertManagement(mgr, namespace, certsReady), "unable to setup cert-controller")
 
 	OpenShift := isOpenShift(ctx, kubeClient.DiscoveryClient)
 
-	if OpenShift {
-		// TODO: setup the RayCluster webhook on vanilla Kubernetes
-		exitOnError(controllers.SetupRayClusterWebhookWithManager(mgr, cfg.KubeRay), "error setting up RayCluster webhook")
-	}
+	go setupControllers(mgr, kubeClient, cfg, OpenShift, certsReady)
+
+	exitOnError(mgr.AddHealthzCheck(cfg.Health.LivenessEndpointName, healthz.Ping), "unable to set up health check")
```

Comment:
Do you want to grab the code block from the AppWrapper controller that defers the health/readiness probes until the certificates or ready?  I called it in #491, but probably best to copy it to the codeflare operator's main.  Code is here: https://github.com/project-codeflare/appwrapper/blob/2213dec59688ddc2259f2d0c5cea4ad759b0e047/pkg/controller/setup.go#L96-L112

Reply:
Right, that's a good idea. Added in af8f912f2218f3a1356a880dbc44af8c592950e1.